### DISCUSSION TITLE
IDispatcher.Dispatch returns bool vs IDispatch.DispatchAsync extension method is void (Task), why?

### DISCUSSION THREAD

**Discussion by mosessaur** [dotnet/maui#14915] [Q&A]
I noticed That `IDispatcher.Dispatch` method returns boolean. And documentation makes sense "true when the action has been dispatched successfully, otherwise false"

On the other hand `IDispatcher.DispatchAsync` doesn't have a return value. Still the documentation says "true if the operation in funcTask was dispatched successfully, otherwise false"

Is the documentation wrong regarding DispatchAsync?

**Comment by pictos** [comment]
Well... `DispatchAsync` isn't on the interface `IDispatacher`, there's no such method there. The DispatchAsync method is a DispatchExtension, as you can see [here](https://github.com/dotnet/maui/blob/0c03fdad7019038071263f1ac380515833f28692/src/Core/src/Dispatching/DispatcherExtensions.cs#L19).

Because of that the return type is different.

Friendly reminder that .net maui is open source, take advantage of that and explore (;

**Reply by mosessaur** [reply]
I understand that is an Extension Method.
But the documentation of the extension method is still incorrect.
If you check the documentation of `public static Task DispatchAsync(this IDispatcher dispatcher, Action action)` it says "true if the operation in action was dispatched successfully, otherwise false"
and for `public static Task DispatchAsync(this IDispatcher dispatcher, Func<Task> funcTask)` it says "true if the operation in funcTask was dispatched successfully, otherwise false"

Thanks for the source link, I was looking for it.

**Reply by pictos** [reply]
Where are the docs? From the inline docs (///) didn't mention the boolean value

**Reply by mosessaur** [reply]
![image](https://user-images.githubusercontent.com/1555270/236112105-64d84281-8a75-401a-bd53-2fc31a65180b.png)
